share/spice/timer/timer.js

```diff
@@ -406,19 +440,31 @@ License: CC BY-NC 3.0 http://creativecommons.org/licenses/by-nc/3.0/
     });
 
     env.ddg_spice_timer = function(api_result) {
+        var timers = [],
+            timerInterval,
+            $addTimerBtn;
+
+        function addTimer(startingTime) {
+            var timer = new Timer(timers.length + 1, startingTime);
+            timer.$element.insertBefore($addTimerBtn.parent());
+            timers.push(timer);
+        }
         
         function onShow() {
```

Comment:
@moollaza 
the purpose of `onShow` is to run whenever the tab is shown. it should never not be used to load more data, but doing something only once isn't exactly an engineering feat. If that's a common pattern, we can control for it. but `onShow` will always fire when it's time to, for example, restart your animation or resume playing a video.

Care to add anything to that @bsstoner ?